### PR TITLE
Feature/user entities and roles

### PR DESCRIPTION
**Pull request by pcsiszar**
Added roles and user repository. (String [] is returned due to vararg contract for hasAnyAuthority())
Users refactored into composition.
Added SampleUsers for a set of initial users.
Added a requestCache based on other Vaadin sample projects.

### CODE REVIEW

File: src/main/java/tk/ynvaser/quiz/model/users/Role.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.stream.Stream;
+
+public enum Role {
+    ADMIN,
+    GAME_MASTER,
+    TEAM_LEADER,
+    USER;
+
+    public static String[] getRoles() {
```
Comment (ynvaser) [inline src/main/java/tk/ynvaser/quiz/model/users/Role.java:11]: I guess this is fine given the use case, but normally we'd return a list.

File: src/main/java/tk/ynvaser/quiz/persistence/entity/UserEntity.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@Entity
+@Data
+public class UserEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @NotEmpty
+    @NotBlank
+    @Pattern(regexp = "^\\w{5,29}$")
```
Comment (ynvaser) [inline src/main/java/tk/ynvaser/quiz/persistence/entity/UserEntity.java]: I'd allow a min length of 3.

File: src/main/java/tk/ynvaser/quiz/security/UserDetailsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public UserDetailsServiceImpl(UserRepository userRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+        if (userRepository.count() == 0) {
+            initUsers();
+        }
+    }
+
+    private void initUsers() {
+        userRepository.saveAll(SampleUsers.getSampleUsers(passwordEncoder));
```
Comment (ynvaser) [inline src/main/java/tk/ynvaser/quiz/security/UserDetailsServiceImpl.java:35]: This is fine for now, but we'll need something like Liquibase to init our DB.

**Review by ynvaser — DISMISSED** [review]

File: src/main/java/tk/ynvaser/quiz/persistence/entity/UserEntity.java (hunk 2 above)
Comment (pcsiszar) [reply src/main/java/tk/ynvaser/quiz/persistence/entity/UserEntity.java]: I touched up on this part to make it 3

**Review by ynvaser — APPROVED** [review]
Looks good! :)

**Pull request merged**